test/management_tests/management_test.py

```diff
@@ -273,12 +275,30 @@ def send_responses(index):
         )
     )
     await asyncio.sleep(0)
-    send_responses(1)
 
     if rate_limit > 0:
-        # this should raise a timeout, since the data was not yet sent
+        # this should raise a timeout, since the data was never confirmed
         with pytest.raises(ManagementConnectionTimeout):
+            # the request is queued
+            await time_travel(0.5 / rate_limit)
+            assert xknx.cemi_handler.send_telegram.call_args_list == []
+
+            # the request should now be sent
+            await time_travel(0.6 / rate_limit)
+
+            assert xknx.cemi_handler.send_telegram.call_args_list == [
+                call(
+                    Telegram(
+                        destination_address=ia,
+                        tpci=tpci.TDataConnected(1),
+                        payload=apci.DeviceDescriptorRead(descriptor=0),
+                    )
+                ),
+            ]
+
             await task
     else:
         # should work as expected
+        send_responses(1)
+
```

Comment:
The test now takes 6 seconds (you can check that by calling  `pytest --durations=10`) waiting for L299 `await task` since before we only make sure that the request is sent, but the timeout only starts then.
But we don't need to test the timeout here - that's done in a different test already. It can be refactored to only test the rate limit and be finished when the responses did arrive - just like the 0-rate_limit parameter does.
```py
    if rate_limit > 0:
        await time_travel(0.5 / rate_limit)
        # the request is still queued
        assert xknx.cemi_handler.send_telegram.call_args_list == []

        await time_travel(0.5 / rate_limit)
        # the request should now be sent
        assert xknx.cemi_handler.send_telegram.call_args_list == [
            call(
                Telegram(
                    destination_address=ia,
                    tpci=tpci.TDataConnected(1),
                    payload=apci.DeviceDescriptorRead(descriptor=0),
                )
            ),
        ]

    send_responses(1)

    await task
```